partiql-ast/src/ast.rs

```diff
@@ -470,13 +470,13 @@ pub struct Path {
 #[derive(Clone, Debug, PartialEq)]
 pub struct Call {
     pub func_name: SymbolPrimitive,
-    pub args: Vec<Box<Expr>>,
+    pub args: Option<Vec<Box<Expr>>>,
 }
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct CallAgg {
     pub func_name: SymbolPrimitive,
-    pub args: Vec<Box<Expr>>,
+    pub args: Option<Vec<Box<Expr>>>,
```

Comment:
To confirm my understanding, `CallAgg` refers to an aggregation function call (like `SUM`, `COUNT`, `AVG`, etc?)?

From the SQL92 BNF (https://ronsavage.github.io/SQL/sql-92.bnf.html#general%20set%20function, can also see pg. 123 of [sql92 spec](http://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt)), the aggregation functions appear to only take one argument, so should this be

```rust
pub arg: Box<Expr>,
```

Reply:
The problem is that the SQL spec formulates the BNF so that there isn't ambiguity as to what is an aggregation function and what is a scalar function. But, in practice, implementing a parser is implemented that was is impractical (the number of rules is crazy).

Instead, here, we parse everything as a function call and will patch things up later when we do type check/inference.

Also, we need to be able to support the syntax if a future integrator adds their own multivalued aggregation function via a plugin interface.

But also, we need the AST to be able to represent what's parseable, even if it's wrong. I think I'd rather parse text like `SELECT COUNT(foo, bar)` and be able prepare an error about the aggregation call than to just throw an unexpected `,`.

Reply:
I see. Thanks for the explanation.